src/read.c

```diff
@@ -5773,7 +5779,47 @@ static avifResult avifDecoderFindGainMapItem(const avifDecoder * decoder,
     return AVIF_RESULT_OK;
 }
 
-static avifResult aviDecoderCheckGainMapProperties(avifDecoder * decoder, const avifPropertyArray * gainMapProperties)
+static avifResult avifDecoderCheckAlphaProperties(avifDecoder * decoder, const avifPropertyArray * alphaProperties)
+{
+    const avifImage * image = decoder->image;
+    // The 'clap', 'irot' and 'imir' transformative properties should be applied to the alpha
+    // auxiliary image item before considering it a plane of the color image item.
+    // Otherwise, inequality with the transformative properties attached to the color image item
+    // should be treated as AVIF_RESULT_NOT_IMPLEMENTED.
+    // The latter is easier and is the behavior of libavif.
+
+    // However, libavif up to version 1.3.0 generated images lacking transformative property
+    // associations with alpha auxiliary image items, so be lenient on their absence for
+    // backward compatibility with previously generated images.
+    const avifProperty * clapProp = avifPropertyArrayFind(alphaProperties, "clap");
+    const avifProperty * irotProp = avifPropertyArrayFind(alphaProperties, "irot");
+    const avifProperty * imirProp = avifPropertyArrayFind(alphaProperties, "imir");
+    if (clapProp == NULL && irotProp == NULL && imirProp == NULL) {
+        return AVIF_RESULT_OK;
+    }
+
+    // TODO(yguyon): Check for 'ispe' values too.
+
+    if (!clapProp != !(image->transformFlags & AVIF_TRANSFORM_CLAP) ||
+        (clapProp && (clapProp->u.clap.widthN != image->clap.widthN || clapProp->u.clap.widthD != image->clap.widthD ||
+                      clapProp->u.clap.heightN != image->clap.heightN || clapProp->u.clap.heightD != image->clap.heightD ||
+                      clapProp->u.clap.horizOffN != image->clap.horizOffN || clapProp->u.clap.horizOffD != image->clap.horizOffD ||
+                      clapProp->u.clap.vertOffN != image->clap.vertOffN || clapProp->u.clap.vertOffD != image->clap.vertOffD))) {
+        avifDiagnosticsPrintf(&decoder->diag, "Clean aperture property mismatch between alpha auxiliary image item and color item");
+        return AVIF_RESULT_NOT_IMPLEMENTED;
```

Comment:
Similar code in the `avifDecoderCheckGainMapProperties()` function returns the `AVIF_RESULT_DECODE_GAIN_MAP_FAILED` error code under this condition. The `AVIF_RESULT_NOT_IMPLEMENTED` error code here seems to imply that this condition isn't really an error; it's just that we haven't written the code to handle this condition properly. Is that what you meant?

Reply:
Yes. This PR focuses on alpha, excepted for https://github.com/AOMediaCodec/libavif/pull/2832#discussion_r2159678461. What should `avifDecoderCheckGainMapProperties()` return is out of scope. Maybe it should be changed to `AVIF_RESULT_NOT_IMPLEMENTED` in another PR. @maryla-uc thoughts?